src/common/dao/label.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dao
+
+import (
+	"time"
+
+	"github.com/astaxie/beego/orm"
+	"github.com/vmware/harbor/src/common/models"
+)
+
+// AddLabel creates a label
+func AddLabel(label *models.Label) (int64, error) {
+	now := time.Now()
+	label.CreationTime = now
+	label.UpdateTime = now
+	return GetOrmer().Insert(label)
```

Comment:
Could you Insert and Query again if the id is zero and error is `nil`?  For compatibility.